cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/checker/SimulationState.java

```diff
@@ -12,119 +11,149 @@
+                    sb.append('*').append(expr);
                 }
+            }
 
-                node.visitAll(new NodeVisitorAdapter() {
-                    @Override
-                    public void accept(ExpressionNode node) {
-                        processExpression(node, suspiciousPoints);
-                    }
-
-                    @Override
-                    public void accept(ArgumentNode node) {
-                        // not to hardcore?
-                        String argument = node.getArgument();
-                        processUsage(argument, suspiciousPoints);
-                    }
-                });
+            @Override
+            public void accept(ConstantNode node) {
+                String value = node.getValue();
+                if (value.endsWith("\n")) {
+                    value = value.replace("\n", ""); // todo this hack should be replaced by fix in parser!
+                }
+                sb.append(value);
             }
         });
+
+        return sb.toString();
     }
 
-    private void processExpression(ExpressionNode node, Set<Node> suspiciousPoints) {
-        if (node.getKey() != null) {
-            return;
-        }
+    void processUsage(ArgumentNode argumentNode) {
+        getValue(argumentNode);
 
-        // fixme dirty
-        String expression = node.getExpression();
-        int expressionStart = expression.indexOf('{');
-        int expressionEnd = expression.lastIndexOf('}');
+        processUsage(getArgument(argumentNode));
+    }
+
+    protected void processUnknownUsage(String variable) {
 
-        if (expressionStart != -1 && expressionEnd != -1) {
-            String argument = expression.substring(expressionStart + 1, expressionEnd);
-            processUsage(argument, suspiciousPoints);
-        }
     }
 
-    private void processUsage(String argument, Set<Node> suspiciousPoints) {
-        Set<CommandInvocationNode> commandInvocationNode = variables.get(argument);
+    void processUsage(String variable) {
+        Set<CommandInvocationNode> nodes = variables.get(variable);
 
-        if (commandInvocationNode == null) {
-            return;
+        if (nodes == null) {
+            processUnknownUsage(variable);
+        } else {
+            suspiciousPoints.removeAll(nodes);
         }
+    }
+
+    void putValue(ArgumentNode argumentNode, CommandInvocationNode command, boolean parentScope) {
+        putValue(argumentNode, command, Collections.emptyMap(), parentScope);
+    }
+
+    void putValue(ArgumentNode argumentNode, CommandInvocationNode command, Map<String, Node> substitutions, boolean parentScope) {
+        // todo check that expressions from argumentNode are used
+
+        // todo inline argumentNode via processUsage()
+        String variable = getValue(argumentNode, substitutions);
+
+        putValue(command, variable, parentScope);
+    }
+
+    protected void putValue(CommandInvocationNode command, String variable, boolean parentScope) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "parentScope". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)